src/chainstate/burn/db/sortdb.rs

```diff
@@ -950,17 +968,24 @@ impl <'a> SortitionHandleConn <'a> {
         SortitionDB::get_block_commit_parent(self, block_height, vtxindex, &self.context.chain_tip)
     }
 
+    /// Get a block commit by txid. In the event of a burnchain fork, this may not be unique.
+    ///   this function simply returns one of those block commits: only use data that is
+    ///   immutable across burnchain/pox forks, e.g., parent block ptr,  
+    pub fn get_block_commit_by_txid(&self, txid: &Txid) -> Result<Option<LeaderBlockCommitOp>, db_error> {
+        let qry = "SELECT * FROM block_commits WHERE txid = ?1 LIMIT 1";
+        query_row(&self.conn, qry, &[&txid])
+    }
+
     /// Get a block commit by its content-addressed location.  Note that burn_header_hash is enough
     /// to identify the fork we're on, since block hashes are globally-unique (w.h.p.) by
     /// construction.
     pub fn get_block_commit(&self, txid: &Txid, burn_header_hash: &BurnchainHeaderHash) -> Result<Option<LeaderBlockCommitOp>, db_error> {
-        // PoX TODO: note -- block_commits table will index on burn_header_hash: if a block_commit is reprocessed due to a PoX fork,
-        //                   it should be allowed to either overwrite the previous entry OR skip insertion (i.e., UNIQUE constraints
-        //                   should not be allowed to cause a panic)
         let qry = "SELECT * FROM block_commits WHERE txid = ?1 AND burn_header_hash = ?2";
         let args: [&dyn ToSql; 2] = [&txid, &burn_header_hash];
-        query_row_panic(&self.conn, qry, &args,
-                        || format!("FATAL: multiple block commits for {},{}", &txid, &burn_header_hash))
+        // note: it's okay just grab the first result if there are multiple entries (because of PoX forks)
+        //       because all the content in the LeaderBlockCommitOp is content-addresses (i.e., the sortition_id isn't
+        //       included in that struct)
```

Comment:
It is possible that this method grabs a block-commit that isn't on the PoX fork, though. Is that okay?

Reply:
In current usage, yes. This method is only ever used once in the code base outside of tests (in `/src/chainstate/burn/sortition.rs`), so it may just be preferable to delete this method and change those instances to use the fetch-by-txid method.